### PR TITLE
Columns fix for blog and work pages, and utility class overhaul

### PR DESCRIPTION
**Pull request by nmorduch**
@harrislapiroff @emilyhorsman take note especially

* I got rid of `margin-leader` and `margin-trailer` and `padding-dbl` and their friends, and replaced with [Bootstrap's versions](https://github.com/twbs/bootstrap/blob/v4-dev/docs/components/utilities.md#spacing)
* Anything like `m-t-1 m-b-1` I consolidated to the equivalent, e.g. `m-y-1`
* Bootstrap [changed](http://v4-alpha.getbootstrap.com/components/utilities/#text-alignment) to `text-xs-center` so we're doing it like that
* Also just some cleanup

Going to the Bootstrap classes means that I squished the old margin system to Bootstrap's `1*spacer` `1.5*spacer` `3*spacer`. If we want to go back to having `2*$spacer` available, or having a `4px` option, we can build those into the $spacer system as described in the [spacing docs](https://github.com/twbs/bootstrap/blob/v4-dev/docs/components/utilities.md#spacing)

### CODE REVIEW

File: core/static/css/modules/_margins.sass [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.m-x-3-1-1-3
```
Comment (emilyhorsman) [inline core/static/css/modules/_margins.sass]: Confusing name?

File: core/templates/core/_blog_summary_text.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
-<header class="blog-post-header margin-trailer-0">
+<header class="blog-post-header m-b-0">
 	<h2 class="blog-post-title card-title">
 		<a href="{{ entry.url }}" class="color-black-almost">{{ entry.title }}</a>
 	</h2>
-	<div class="blog-post-byline-dateline margin-trailer-tiny">
+	<div class="blog-post-byline-dateline m-b-1">
```
Comment (emilyhorsman) [inline core/templates/core/_blog_summary_text.html]: :+1:

**Comment by emilyhorsman** [comment]
Looks good to me other than https://github.com/littleweaver/littleweaverweb.com/pull/60#discussion_r54750051

:sparkles:

File: core/static/css/modules/_margins.sass (hunk 1 above)
Comment (nmorduch) [inline core/static/css/modules/_margins.sass]: I know :( But I can't think of a better one that is not entirely abstract.

File: core/templates/core/_blog_summary_text.html (hunk 2 above)
Comment (nmorduch) [inline core/templates/core/_blog_summary_text.html]: Yeah we could make a class for this if we want. I don't really care either way

**Comment by harrislapiroff** [comment]
Looks good. And I'm okay enough with the confusing class name.

**Pull request merged**